string_grouper/string_grouper.py

```diff
@@ -711,3 +942,141 @@ def _validate_id_data(master, duplicates, master_id, duplicates_id):
             raise Exception('Both master and master_id must be pandas.Series of the same length.')
         if duplicates is not None and duplicates_id is not None and len(duplicates) != len(duplicates_id):
             raise Exception('Both duplicates and duplicates_id must be pandas.Series of the same length.')
+
+
+# High level functions
+
+
+def compute_pairwise_similarities(string_series_1: pd.Series,
+                                  string_series_2: pd.Series,
+                                  corpus: Optional[StringGrouper] = None,
+                                  **kwargs) -> pd.Series:
+    """
+    Computes the similarity scores between two Series of strings row-wise.
+
+    :param string_series_1: pandas.Series. The input Series of strings to be grouped
+    :param string_series_2: pandas.Series. The input Series of the IDs of the strings to be grouped
+    :param corpus: An already initialized StringGrouper object (thus already containing a corpus).
+    :param kwargs: All other keyword arguments are passed to StringGrouperConfig
+    :return: pandas.Series of similarity scores, the same length as string_series_1 and string_series_2
+    """
+    if not corpus:
+        corpus = StringGrouper(string_series_1, string_series_2, **kwargs)
+    else:
+        corpus.reset_data(string_series_1, string_series_2)
```

Comment:
I'm not sure we should the corpus option to the higher level functions. Basically this is an StringGrouper with a pre-fitted tf-idf vectorizer right? I think there are some dangers here (mainly - the new series can contain ngrams that are not known in the vectorizer), that one should know about when using this. Also it implies deeper knowledge of how the StringGrouper object works. The "High level functions" were created to be able to use the string grouper class without much reading into the code. Once you start inserting your own tf-idf vectorizer I think you are fairly knowledgeable on how the class works, and should use the class functions directly. What do you think? 

I guess i'm not seeing a (common) situation where a user is first building a StringGrouper, and then inserting that in one of the High Level functions, but maybe I'm missing something